GithubService.Services/Converters/CodeConverter.cs

```diff
@@ -0,0 +1,100 @@
+﻿using GithubService.Models;
+using GithubService.Models.KenticoCloud;
+using GithubService.Services.Interfaces;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace GithubService.Services.Converters
+{
+    public class CodeConverter : ICodeConverter
+    {
+        public IEnumerable<CodenameCodeFragments> ConvertToCodenameCodeFragments(IEnumerable<CodeFragment> fragments)
+        {
+            var codenameCodeFragments = new Dictionary<string, CodenameCodeFragments>();
+
+            foreach (var codeFragment in fragments)
+            {
+                var codename = codeFragment.Codename;
+                var language = codeFragment.Language;
+                var content = codeFragment.Content;
+
+                if (codenameCodeFragments.ContainsKey(codename))
+                {
+                    codenameCodeFragments[codename].CodeFragments.Add(language, content);
+                }
+                else
+                {
+                    codenameCodeFragments.Add(codename, new CodenameCodeFragments
+                    {
+                        Codename = codename,
+                        CodeFragments = new Dictionary<CodeFragmentLanguage, string>
+                            {
+                                {language, content}
+                            }
+                    });
+                }
+            }
+
+            return codenameCodeFragments.Values;
+        }
+
+        public CodeSamples ConvertToCodeSamples(CodenameCodeFragments codenameCodeFragments) => new CodeSamples
+        {
+            Curl = GetLanguageContent(CodeFragmentLanguage.CUrl, codenameCodeFragments),
+            CSharp = GetLanguageContent(CodeFragmentLanguage.CSharp, codenameCodeFragments),
+            JavaScript = GetLanguageContent(CodeFragmentLanguage.JavaScript, codenameCodeFragments),
+            TypeScript = GetLanguageContent(CodeFragmentLanguage.TypeScript, codenameCodeFragments),
+            Java = GetLanguageContent(CodeFragmentLanguage.Java, codenameCodeFragments),
+            JavaRx = GetLanguageContent(CodeFragmentLanguage.JavaRx, codenameCodeFragments),
+            PHP = GetLanguageContent(CodeFragmentLanguage.PHP, codenameCodeFragments),
+            Swift = GetLanguageContent(CodeFragmentLanguage.Swift, codenameCodeFragments),
+            Ruby = GetLanguageContent(CodeFragmentLanguage.Ruby, codenameCodeFragments)
+        };
+
+        public (List<CodeFragment> newFragments, List<CodeFragment> modifiedFragments, List<CodeFragment> removedFragments) CompareFragmentLists(List<CodeFragment> oldFragmentList, List<CodeFragment> newFragmentList)
```

Comment:
Do we need an extra list for newFragments and modifiedFragments? In Update function, we use both of them together in fragmentsToUpsert.